internal/services/synapse/synapse_workspace_resource.go

```diff
@@ -952,7 +960,15 @@ func flattenIdentity(input *synapse.ManagedIdentity) []interface{} {
 		if input.TenantID != nil {
 			config.TenantId = input.TenantID.String()
 		}
+		identityIds := make(map[string]identity.UserAssignedIdentityDetails)
+		for k := range input.UserAssignedIdentities {
+			identityIds[k] = identity.UserAssignedIdentityDetails{
+				// since v is an `interface{}` there's no guarantee this is returned
```

Comment:
that would indicate a Swagger issue, is there an issue tracking this? otherwise we should pass in Client ID and Principal ID here